modules/honzon-distribution/src/lib.rs

```diff
@@ -0,0 +1,329 @@
+pub enum DistributionDestination<AccountId> {
+	StableAsset(DistributionToStableAsset<AccountId>),
+}
+
+/// Information needed when distribution to StableAsset.
+#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct DistributionToStableAsset<AccountId> {
+	pub pool_id: StableAssetPoolId,
+	pub stable_token_index: u32,
+	pub stable_currency_id: CurrencyId,
+	pub account_id: AccountId,
+}
+
+/// Distribution params
+#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq, Default, TypeInfo, MaxEncodedLen)]
+pub struct DistributionParams {
+	pub capacity: Balance,
+	pub max_step: Balance,
+	pub target_min: Ratio,
+	pub target_max: Ratio,
+}
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Adjust time period.
+		type AdjustPeriod: Get<Self::BlockNumber>;
+
+		/// Taiga stable asset protocol.
+		type StableAsset: StableAssetT<
+			AssetId = CurrencyId,
+			AtLeast64BitUnsigned = Balance,
+			Balance = Balance,
+			AccountId = Self::AccountId,
+			BlockNumber = Self::BlockNumber,
+		>;
+
+		/// Currency for deposit/withdraw assets.
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>;
+
+		/// The origin updating params and force adjust.
+		type UpdateOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Weight information for the extrinsics in this module.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn distribution_destination_params)]
+	pub type DistributionDestinationParams<T: Config> =
+		StorageMap<_, Twox64Concat, DistributionDestination<T::AccountId>, DistributionParams, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn distributed_balance)]
+	pub type DistributedBalance<T: Config> =
+		StorageMap<_, Twox64Concat, DistributionDestination<T::AccountId>, Balance, OptionQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The DistributionParams is not exist.
+		DistributionParamsNotExist,
+		/// Exceed capacity
+		ExceedCapacity,
+		/// The Destination is invalid.
+		InvalidDestination,
+		/// The DistributionParams is invalid
+		InvalidDistributionParams,
+		/// The balance is invalid
+		InvalidUpdateBalance,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub fn deposit_event)]
+	pub enum Event<T: Config> {
+		UpdateDistributionParams {
+			destination: DistributionDestination<T::AccountId>,
+			params: DistributionParams,
+		},
+		AdjustDestination {
+			destination: DistributionDestination<T::AccountId>,
+			stable_currency: CurrencyId,
+			amount: Amount,
+		},
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			if now % T::AdjustPeriod::get() == Zero::zero() {
```

Comment:
I want to make it `now % period === offset` so that we can purposely ensure the `on_initialize` hooks for different pallets won't be all scheduled on a same block

Reply:
What's the production runtime value should be? current I'm setting period=DAYS and offset=MINUTES.

Reply:
period = 1 day, offset = 1 block